src/asm_file.py

```diff
@@ -358,24 +368,25 @@ def process_label(label: str, *, glabel: bool) -> None:
         if not line:
             continue
 
-        if line.startswith("."):
+        directive = line.split()[0]
+        if directive.startswith("."):
             # Assembler directive.
-            if line.startswith(".ifdef") or line.startswith(".ifndef"):
+            if directive == ".ifdef" or directive == ".ifndef":
                 macro_name = line.split()[1]
                 if macro_name not in defines:
-                    defines[macro_name] = 0
+                    defines[macro_name] = None
                     add_warning(
                         warnings,
                         f"Note: assuming {macro_name} is unset for .ifdef, "
                         f"pass -D{macro_name}/-U{macro_name} to set/unset explicitly.",
                     )
-                level = defines[macro_name]
-                if line.startswith(".ifdef"):
+                level = 1 if defines[macro_name] is not None else 0
+                if directive == ".ifdef":
                     level = 1 - level
                 ifdef_level += level
                 ifdef_levels.append(level)
-            elif line.startswith(".if"):
-                macro_name = line.split()[1]
+            elif directive.startswith(".if"):
+                macro_name = directive.split()[1]
```

Comment:
I think this should stay `macro_name = line.split()[1]`? This breaks on some existing files.

Reply:
Oops! Good catch